whois-nrtm4/src/main/java/net/ripe/db/nrtm4/SnapshotFileGenerator.java

```diff
@@ -3,92 +3,148 @@
+import net.ripe.db.whois.common.domain.CIString;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Service;
 
-import java.io.IOException;
-import java.io.OutputStream;
+import java.io.ByteArrayOutputStream;
 import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Map;
 import java.util.Optional;
+import java.util.Set;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.zip.GZIPOutputStream;
 
 
 @Service
 public class SnapshotFileGenerator {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(SnapshotFileGenerator.class);
+    private static final int QUEUE_CAPACITY = 1000;
 
+    private final String whoisSource;
+    private final NrtmFileService nrtmFileService;
     private final NrtmVersionInfoRepository nrtmVersionInfoRepository;
-    private final SnapshotFileSerializer snapshotFileSerializer;
+    private final RpslObjectEnqueuer rpslObjectEnqueuer;
     private final SnapshotFileRepository snapshotFileRepository;
-    private final SnapshotObjectSynchronizer snapshotObjectSynchronizer;
+    private final SnapshotFileSerializer snapshotFileSerializer;
     private final SourceRepository sourceRepository;
-    private final NrtmFileStore nrtmFileStore;
+    private final WhoisObjectRepository whoisObjectRepository;
 
     public SnapshotFileGenerator(
+        @Value("${whois.source}") final String whoisSource,
+        final NrtmFileService nrtmFileService,
         final NrtmVersionInfoRepository nrtmVersionInfoRepository,
-        final SnapshotFileSerializer snapshotFileSerializer,
+        final RpslObjectEnqueuer rpslObjectEnqueuer,
         final SnapshotFileRepository snapshotFileRepository,
-        final SnapshotObjectSynchronizer snapshotObjectSynchronizer,
+        final SnapshotFileSerializer snapshotFileSerializer,
         final SourceRepository sourceRepository,
-        final NrtmFileStore nrtmFileStore
+        final WhoisObjectRepository whoisObjectRepository
     ) {
+        this.whoisSource = whoisSource;
+        this.nrtmFileService = nrtmFileService;
         this.nrtmVersionInfoRepository = nrtmVersionInfoRepository;
-        this.snapshotFileSerializer = snapshotFileSerializer;
+        this.rpslObjectEnqueuer = rpslObjectEnqueuer;
         this.snapshotFileRepository = snapshotFileRepository;
-        this.snapshotObjectSynchronizer = snapshotObjectSynchronizer;
+        this.snapshotFileSerializer = snapshotFileSerializer;
         this.sourceRepository = sourceRepository;
-        this.nrtmFileStore = nrtmFileStore;
+        this.whoisObjectRepository = whoisObjectRepository;
     }
 
-    public List<PublishableSnapshotFile> createSnapshots() throws IOException {
+    public Set<PublishableNrtmFile> createSnapshots() {
```

Comment:
this side of code is a bit complicated to understand.
Could we split into modules? which a descriptive name per each module?

Reply:
Thank you! It is easier to read now